index.js

```diff
@@ -295,7 +295,13 @@ export default class Emittery {
 			}
 		}
 
-		return this.off.bind(this, eventNames, listener);
+		const off = this.off.bind(this, eventNames, listener);
+
+		if (options?.signal instanceof globalThis.AbortSignal) {
+			options.signal.addEventListener('abort', off);
+		}
```

Comment:
```suggestion
		options?.signal?.addEventListener('abort', off);
```

Reply:
You should also not pass the `off` method directly in.

Reply:
> You should also not pass the `off` method directly in.

If not, what should we do?

Reply:
No longer relevant.